api/usecases_with_context_credentials.go

```diff
@@ -3,49 +3,13 @@ package api
 import (
 	"fmt"
 	"net/http"
-	"time"
-
-	"github.com/gin-gonic/gin"
-	"golang.org/x/net/http2"
-	"golang.org/x/net/http2/h2c"
 
 	"github.com/checkmarble/marble-backend/models"
 	"github.com/checkmarble/marble-backend/usecases"
 	"github.com/checkmarble/marble-backend/utils"
 )
 
-type API struct {
-	router        *gin.Engine
-	usecases      usecases.Usecases
-	marbleAppHost string
-}
-
-func New(
```

Comment:
This was horrible spaghetti code with unclear dependencies, and I was left befuddled several times when I tried who of api, router, server did what.
I cleaned it up to delete the API struct. Dependencies (in particular usecases) are now injected at the route definition level. See routes.go. The really relevant changes are in there.